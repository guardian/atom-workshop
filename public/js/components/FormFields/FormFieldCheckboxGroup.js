import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';
import FormFieldCheckbox from './FormFieldCheckbox';

export default class FormFieldCheckboxGroup extends React.Component {

  static propTypes = {

    fieldLabel: PropTypes.string.isRequired,
    fieldName: PropTypes.string.isRequired,
    fieldValue: PropTypes.array.isRequired,
    checkValues: PropTypes.array.isRequired,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    onUpdateField: PropTypes.func.isRequired
  };

  isChecked = (checkValue) => {
    return this.props.fieldValue.includes(checkValue);
  }

  renderCheckbox(fieldName, i) {
    const updateFn = (newValue) => {
      let newFieldValue = [];
      if(newValue && !this.isChecked(newValue)) {
        newFieldValue = this.props.fieldValue.concat([fieldName]);
      } else {
        newFieldValue = this.props.fieldValue.filter((oldFieldName) => {
          return fieldName !== oldFieldName;
        });
      }
      this.props.onUpdateField(newFieldValue);
    };

    return (
      <FormFieldCheckbox
        key={`${this.props.fieldName}-${i}`}
        fieldName={`${this.props.fieldName}-${i}`}
        fieldValue={this.isChecked(fieldName)}
        onUpdateField={updateFn}
        checkGroup="true" />
    );
  }

  render() {
    return (
        <div>
          <label className="form__label" htmlFor={this.props.fieldName}>{this.props.fieldLabel}</label>
          {this.props.checkValues.map((fieldName, i) => this.renderCheckbox(fieldName, i))}
          <ShowErrors errors={this.props.fieldErrors}/>
        </div>
    );
  }
}
