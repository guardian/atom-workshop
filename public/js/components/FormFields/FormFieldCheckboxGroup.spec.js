import React from 'react';
import FormFieldCheckboxGroup from './FormFieldCheckboxGroup';
import renderer from 'react-test-renderer';

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue = ['one', 'two'],
    checkValues = ['one', 'two', 'three'];

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldCheckboxGroup fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} checkValues={checkValues} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
