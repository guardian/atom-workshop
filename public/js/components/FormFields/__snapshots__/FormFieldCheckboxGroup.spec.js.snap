exports[`test Should render 1`] = `
<div>
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <div
    className="form__group form__group--checkbox">
    <input
      checked={true}
      className="form__checkbox"
      name="test-0"
      onChange={[Function]}
      type="checkbox"
      value={true} />
    <span
      className="form__label form__label--checkbox">
      test-0
    </span>
    <div />
  </div>
  <div
    className="form__group form__group--checkbox">
    <input
      checked={true}
      className="form__checkbox"
      name="test-1"
      onChange={[Function]}
      type="checkbox"
      value={true} />
    <span
      className="form__label form__label--checkbox">
      test-1
    </span>
    <div />
  </div>
  <div
    className="form__group form__group--checkbox">
    <input
      checked={false}
      className="form__checkbox"
      name="test-2"
      onChange={[Function]}
      type="checkbox"
      value={false} />
    <span
      className="form__label form__label--checkbox">
      test-2
    </span>
    <div />
  </div>
  <div />
</div>
`;
